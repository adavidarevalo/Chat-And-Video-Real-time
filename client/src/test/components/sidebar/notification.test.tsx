import React from 'react';
import { render, screen } from '@testing-library/react';
import SidebarNotification from '../../../components/sidebar/notification';
import '@testing-library/jest-dom/extend-expect';

describe('SidebarNotification', () => {
  it('should render correctly', () => {
    const { container } = render(<SidebarNotification />);

    expect(container).toMatchSnapshot();
    expect(screen.getByTestId('notification-icon')).toBeInTheDocument();
    expect(screen.getByTestId('close-icon')).toBeInTheDocument();
    expect(
      screen.getByText('Get notified of new messages'),
    ).toBeInTheDocument();
    expect(
      screen.getByText('Turn on desktop notification'),
    ).toBeInTheDocument();
  });
});
