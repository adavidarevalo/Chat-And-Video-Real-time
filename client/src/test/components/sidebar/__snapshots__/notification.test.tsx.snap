// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarNotification should render correctly 1`] = `
<div>
  <div
    class="h-[90px] dark:bg-dark_bg_3 flex items-center p-[13px]"
  >
    <div
      class="w-full flex items-center justify-between"
    >
      <div
        class="flex items-center gap-x-4"
      >
        <div
          class="cursor-pointer"
        >
          <div
            data-testid="notification-icon"
          >
            <svg
              enable-background="new 0 0 48 48"
              height="48"
              preserveAspectRatio="xMidYMid meet"
              version="1.1"
              viewBox="0 0 48 48"
              width="48"
              x="0px"
              xml:space="preserve"
              y="0px"
            >
              <path
                class="dark:fill-blue_1"
                d="M24.154,2C11.919,2,2,11.924,2,24.165S11.919,46.33,24.154,46.33 s22.154-9.924,22.154-22.165S36.389,2,24.154,2z M23.41,17.428V16.81c0-0.706,0.618-1.324,1.324-1.324s1.323,0.618,1.323,1.324 v0.618c2.559,0.618,4.412,2.823,4.412,5.559v3.176l-8.294-8.294C22.527,17.692,22.969,17.516,23.41,17.428z M24.733,33.134 c-0.971,0-1.765-0.794-1.765-1.765h3.529C26.498,32.34,25.704,33.134,24.733,33.134z M31.969,32.251l-1.765-1.765H17.233v-0.882 l1.765-1.765v-4.853c0-1.059,0.265-2.029,0.794-2.912l-2.559-2.559l1.147-1.147l14.735,14.736L31.969,32.251z"
              />
            </svg>
          </div>
        </div>
        <div
          class="flex flex-col"
        >
          <span
            class="textPrimary"
          >
            Get notified of new messages
          </span>
          <span
            class="textSecondary mt-0.5 items-center gap-0.5 flex"
          >
            Turn on desktop notification
            <svg
              class="dark:fill-dark_svg_2 mt-1 ml-2"
              enable-background="new 0 0 8 12"
              height="12"
              preserveAspectRatio="xMidYMid meet"
              version="1.1"
              viewBox="0 0 8 12"
              width="8"
              x="0px"
              xml:space="preserve"
              y="0px"
            >
              <path
                class="dark:fill-dark_svg_2 mt-1 ml-2"
                d="M2.173,1l4.584,4.725L2.142,10.34L1.039,9.237l3.512-3.512L1,2.173L2.173,1z"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        class="cursor-pointer"
      >
        <div
          data-testid="close-icon"
        >
          <svg
            class="dark:fill-dark_svg_2"
            enable-background="new 0 0 24 24"
            height="24"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 24 24"
            width="24"
            xml:space="preserve"
          >
            <path
              d="M19.6004 17.2L14.3004 11.9L19.6004 6.60005L17.8004 4.80005L12.5004 10.2L7.20039 4.90005L5.40039 6.60005L10.7004 11.9L5.40039 17.2L7.20039 19L12.5004 13.7L17.8004 19L19.6004 17.2Z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
